molecule/util.py

```diff
@@ -189,20 +191,51 @@ def reset_known_host_key(hostname):
     return os.system('ssh-keygen -R {}'.format(hostname))
 
 
-def check_ssh_availability(hostip, user, timeout):
+def check_ssh_availability(hostip, user, timeout, sshkey_filename):
     import socket
     ssh = paramiko.SSHClient()
     ssh.load_system_host_keys()
     ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
     try:
-        ssh.connect(hostip, username=user)
+        ssh.connect(hostip, username=user, key_filename=sshkey_filename)
         return True
     except (paramiko.BadHostKeyException, paramiko.AuthenticationException,
             paramiko.SSHException, socket.error):
         time.sleep(timeout)
         return False
 
 
+def generated_ssh_key_file_location():
+    return GENERATED_SSH_KEY_LOCATION
+
+
+def generate_temp_ssh_key():
```

Comment:
Does this really belong in util?  Since it is specific to openstack ,and only used int he openstack driver, prob should be relocated to openstack driver class.